### PR TITLE
docs: Add more potential use case for Bottom Navigation Bar

### LINKED ISSUE #33810: [Documentation] Additional BottomNavigationBar use cases through samples
## Use case
Currently, the `BottomNavigationBar` [documentation](https://api.flutter.dev/flutter/material/BottomNavigationBar-class.html) supplies sample code for a simple example involving updating the displayed `Scaffold.body` widget. However, there can also be more unique interaction patterns, such as:
1) tapping an already-selected index for additional behavior (ie. to scroll to top of displayed page)
2) having a unique item that doesn't update the selectedIndex (ie. a button that opens a dropdown or modal instead of updating the displayed `Scaffold.body`)

While these are less common, there is some desire for instruction on getting these behaviors implemented.

## Proposal
Create sample docs for one or both cases above to show that `BottomNavigationBar` can be more flexible than expected.

### PR DESCRIPTION
**Pull request by amanv8060**
Adds the dartpad example for the use case mentioned in the issue

Fixes #33810

*If you had to change anything in the [flutter/tests] repo, include a link to the migration guide as per the [breaking change policy].*

## Pre-launch Checklist

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] I read the [Tree Hygiene] wiki page, which explains my responsibilities.
- [x] I read and followed the [Flutter Style Guide], including [Features we expect every widget to implement].
- [x] I signed the [CLA].
- [x] I listed at least one issue that this PR fixes in the description above.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] I added new tests to check the change I am making, or this PR is [test-exempt].
- [x] All existing and new tests are passing.

If you need help, consider asking for advice on the #hackers-new channel on [Discord].

[Contributor Guide]: https://github.com/flutter/flutter/wiki/Tree-hygiene#overview
[Tree Hygiene]: https://github.com/flutter/flutter/wiki/Tree-hygiene
[test-exempt]: https://github.com/flutter/flutter/wiki/Tree-hygiene#tests
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[Features we expect every widget to implement]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#features-we-expect-every-widget-to-implement
[CLA]: https://cla.developers.google.com/
[flutter/tests]: https://github.com/flutter/tests
[breaking change policy]: https://github.com/flutter/flutter/wiki/Tree-hygiene#handling-breaking-changes
[Discord]: https://github.com/flutter/flutter/wiki/Chat

### CODE REVIEW

**Comment by amanv8060** [comment]
@HansMuller does this pr need additional changes?

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart [hunk 1]
```diff
@@ -118,6 +118,88 @@ enum BottomNavigationBarLandscapeLayout {
...
 /// ** See code in examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.1.dart **
 /// {@end-tool}
+///
+/// {@tool dartpad --template=stateful_widget_material}
+/// This example shows [BottomNavigationBar] used in a [Scaffold] Widget with
+/// different interaction patterns. Tapping on the first [BottomNavigationBarItem]
+/// uses the [ScrollController] to animate the [ListView] to the top. The second
+/// and third [BottomNavigationBarItem] change the fontWeight to bold and normal
+/// respectively.
+/// ```dart
```
Comment (guidezpl) [inline packages/flutter/lib/src/material/bottom_navigation_bar.dart:128]: Example code is now hosted in separate files. Can you please put this code in `examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.2.dart`? You can follow `examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.1.dart` as a model.

**Review by guidezpl — CHANGES_REQUESTED** [review]

File: packages/flutter/lib/src/material/bottom_navigation_bar.dart (hunk 1 above)
Comment (amanv8060) [reply packages/flutter/lib/src/material/bottom_navigation_bar.dart:128]: Sure, will do it.

Comment (amanv8060) [reply packages/flutter/lib/src/material/bottom_navigation_bar.dart]: @guidezpl made the requested changes and added tests.

File: examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.2.dart [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+              icon: Icon(Icons.format_clear),
+              label: 'Normalize Fonts',
+            ),
+          ],
+          currentIndex: _selectedIndex,
+          selectedItemColor: Colors.amber[800],
+          onTap: (int index) {
+            switch (index) {
+              case 0:
+                _homeController.animateTo(
```
Comment (guidezpl) [inline examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.2.dart:83]: This should only scroll if the selected tab is `index`

**Review by guidezpl — CHANGES_REQUESTED** [review]
Thanks for contributing! I like the example for scrolling to top, but using bottom nav to change state is a bit of an anti-pattern. 

It would be more useful to showcase an example like "having a unique item that doesn't update the selectedIndex (ie. a button that opens a dropdown or modal instead of updating the displayed Scaffold.body)"

File: examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.2.dart (hunk 2 above)
Comment (amanv8060) [reply examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.2.dart:77]: changed.

Comment (guidezpl) [reply examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.2.dart:77]: I'm still seeing `_selectedIndex == 0`, perhaps you need to push?

Comment (amanv8060) [reply examples/api/lib/material/bottom_navigation_bar/bottom_navigation_bar.2.dart:77]: Should be fine now.

**Review by guidezpl — APPROVED** [review]

**Review by jonahwilliams — APPROVED** [review]
RSLGTM

**Pull request merged**